src/charm.py

```diff
@@ -205,6 +207,110 @@ def _check_kf_profiles(self, interfaces):
 
         return kf_profiles
 
+    def main(self, event) -> None:
+        """Main entry point for the Charm."""
+        try:
+            self._check_container_connection()
+            self._check_model_name()
+            self._check_leader()
+            interfaces = self._get_interfaces()
+            kf_profiles = self._check_kf_profiles(interfaces)
+            self.handle_ingress(interfaces)
+        except CheckFailed as e:
+            self.model.unit.status = e.status
+            return
+        try:
+            self.unit.status = MaintenanceStatus("Creating k8s resources")
+            self.k8s_resource_handler.apply()
+        except ApiError:
+            self.logger.error(traceback.format_exc())
+            self.unit.status = BlockedStatus("kubernetes resource creation failed")
+            return
+        self.handle_ingress(interfaces)
+        kf_profiles = list(kf_profiles.get_data().values())[0]
+        self.profiles_service = kf_profiles["service-name"]
+        try:
+            self._update_layer()
+        except CheckFailed as e:
+            self.model.unit.status = e.status
+            return
+        self.model.unit.status = ActiveStatus()
+
+    def _get_sidebar_config(self) -> Dict:
+        try:
+            current_configmap = self.k8s_resource_handler.lightkube_client.get(
+                ConfigMap, name=self._context["configmap_name"]
+            )
+            old_sidebar_config = json.loads(current_configmap.data["links"])
+        except ApiError as e:
+            self.logger.warning(
+                f"PROBLEM during configmap retrieval {e}. USING BASE CONFIG"
+            )
+            old_sidebar_config = json.loads(BASE_SIDEBAR)
```

Comment:
When reading this, my first instinct was "if we can't get the configmap, Block or Error" (probably Block, just in case it resolves itself somehow).  I'm not sure the workload functions without that configmap being accessible

Is there a case here where we want to gracefully handle the missing configmap like this, providing instead a default (without sidebar links)?

Reply:
fwiw, I'm not really sure what feels best.  just asking the question :)

Reply:
Resolved in new way of handlig sidebar.